xcube/core/store/stores/memory.py

```diff
@@ -57,32 +57,48 @@ def get_data_store_params_schema(cls) -> JsonObjectSchema:
         return JsonObjectSchema()
 
     @classmethod
-    def get_type_ids(cls) -> Tuple[str, ...]:
-        return str(TYPE_ID_ANY),
+    def get_type_specifiers(cls) -> Tuple[str, ...]:
+        return str(TYPE_SPECIFIER_ANY),
 
-    def get_data_ids(self, type_id: str = None) -> Iterator[Tuple[str, Optional[str]]]:
+    def get_type_specifiers_for_data(self, data_id: str) -> Tuple[str, ...]:
+        self._assert_valid_data_id(data_id)
+        type_specifier = get_type_specifier(self._data_dict[data_id])
+        return str(type_specifier),
+
+    def get_data_ids(self, type_specifier: str = None, include_titles: bool = True) -> Iterator[Tuple[str, Optional[str]]]:
         for data_id, data in self._data_dict.items():
-            if type_id is None or TypeId.normalize(type_id).is_compatible(get_type_id(data)):
+            if type_specifier is None or TypeSpecifier.normalize(type_specifier).is_compatible(get_type_specifier(data)):
                 yield data_id, None
 
-    def has_data(self, data_id: str) -> bool:
+    def has_data(self, data_id: str, type_specifier: str = None) -> bool:
         assert_given(data_id, 'data_id')
-        return data_id in self._data_dict
-
-    def describe_data(self, data_id: str) -> DataDescriptor:
+        if not data_id in self._data_dict:
+            return False
+        if type_specifier:
```

Comment:
```suggestion
        if type_specifier is not None:
```

There are a few things (other than None) that may evaluate to False in this test, so I prefer an explicit None check. Admittedly it's unlikely that we'd be using e.g. 0 or the empty string as a type specifier, but if something like that is passed by mistake I think it's preferable to process it here (probably producing False or an error) rather than falling through to returning True.